### PR TITLE
Retrieve current forgers - Closes #240

### LINKED ISSUE #240: Retrieve current forgers
### Description

Make sure the current round forgers retrieval works properly, using the same techniques that work in the previous versions of the SDK.

Make sure the following search criteria are supported:

Current forgers can be stored in-memory.

### Motivation

- Random access and search by those properties is required by the UI

### Acceptance Criteria

- Retrieval of next forgers is possible
- Gateway is updated with the API version 2

#### Endpoints

- HTTP `/api/v2/forgers`
- RPC `get.forgers`

#### Request parameters

Parameter | Type | Validation | Default | Comment
-- | -- | -- | -- | --
limit | Number | <1;103> | 10 |  
offset | Number | <0;+Inf> | 0 |  

#### Response example

200 OK

Make the version 2 API able to retrieve data by those criteria.

```jsonc
{
  "data": [
    {
      "username": "genesis_51",
      "totalVotesReceived": "1006000000000",
      "address": "c6d076ed541ca20869a1398a9d28c645ac8a8719",
      "minActiveHeight": 27605,
      "isConsensusParticipant": true,
      "nextForgingTime": 1607521557
    },
  ],
  "meta": {
    "count": 10,
    "offset": 20,
    "total": 103
  },
  "links": {}
}

```

400 Bad Request
```jsonc
{
  "error": true,
  "message": "Unknown input parameter(s): <param_name>"
}
```

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #240 

### How was it solved?
- [x] HTTP /api/v2/forgers works
- [x]  RPC get.forgers works
- [x] Retrieval of next forgers is possible
- [x] Gateway is updated with the API version 2
- [x] response map properly
- [x] reload on round/change (app:chain:validators:change)
- [x] reload forgers during startup

### How was it tested?
Local

### CODE REVIEW

File: services/core/shared/core/delegates.js [hunk 1]
```diff
@@ -206,14 +206,21 @@ const getNextForgers = async params => {
...
 
 const resolveNextForgers = async () => {
-	nextForgers = await BluebirdPromise.map(rawNextForgers,
-		async forger => delegateList.find(o => o.address === forger.address));
+	if (sdkVersion <= 4) {
+		nextForgers = await BluebirdPromise.map(rawNextForgers,
+			async forger => delegateList.find(o => o.address === forger.address));
+	} else {
+		nextForgers = rawNextForgers.data;
+	}
```
Comment (sameersubudhi) [inline services/core/shared/core/delegates.js:223]: Please test if this works.

```suggestion
	nextForgers = await BluebirdPromise.map(
		rawNextForgers,
		async forger => sdkVersion <= 4
			? delegateList.find(o => o.address === forger.address)
			: forger,
	);
```

File: services/core/shared/core/compat/sdk_v5/forgers.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// TODO: Update coreApi.getAccount() to account.getAccounts()
+const getForgers = async params => {
+	const forgers = await coreApi.getForgers(params);
+	const forgerAddresses = forgers.data.map(forger => forger.address);
+	const forgerAccounts = await coreApi.getAccounts({ addresses: forgerAddresses });
+	forgers.data = forgers.data.map(forger => {
+		const filterAcc = forgerAccounts.data
+			.filter(account => account.address.toString('hex') === forger.address);
+		forger.username = filterAcc[0].dpos.delegate.username;
+		forger.totalVotesReceived = Number(filterAcc[0].dpos.delegate.totalVotesReceived);
```
Comment (sameersubudhi) [inline services/core/shared/core/compat/sdk_v5/forgers.js:31]:
```suggestion
		const filteredAcc = forgerAccounts.data
			.filter(account => account.address.toString('hex') === forger.address);
		forger.username = filteredAcc[0].dpos.delegate.username;
		forger.totalVotesReceived = Number(filteredAcc[0].dpos.delegate.totalVotesReceived);
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
The implementation looks good. However, please consider my remarks below. They are all good to have.

File: services/core/events/blockchain.js [hunk 3]
```diff
@@ -55,14 +55,30 @@ module.exports = [
...
 				logger.debug('New round, updating delegates...');
-				core.reloadDelegateCache();
-				core.reloadNextForgersCache();
+				// TODO: Remove if condition once delegate cache implemented for SDKv5
+				if (core.getSDKVersion() <= 4) {
+					core.reloadDelegateCache();
+					core.reloadNextForgersCache();
+				} else {
+					core.reloadNextForgersCache();
+				}
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply services/core/events/blockchain.js]: Done, thanks!

File: services/gateway/app.js [hunk 4]
```diff
@@ -121,6 +124,10 @@ broker.createService({
 	events: {
 		'block.change': (payload) => sendSocketIoEvent('update.block', payload),
 		'round.change': (payload) => sendSocketIoEvent('update.round', payload),
+		'forgers.change': (payload) => sendSocketIoEvent('update.forgers', mapper(payload, {
+			data: ['data', delegateResponse],
+			meta: {},
+		})),
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply services/gateway/app.js:130]: Done!

File: services/core/shared/core/delegates.js (hunk 1 above)
Comment (nagdahimanshu) [reply services/core/shared/core/delegates.js]: Done!

File: services/gateway/app.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (MichalTuleja) [reply services/gateway/app.js:130]: Actually, all of them should be doing that, this is why we've got a separate config files.

**Review by MichalTuleja — APPROVED** [review]

**Pull request merged**